SmartDeviceLink/SDLResponseDispatcher.m

```diff
@@ -140,6 +143,43 @@ - (void)clear {
     _audioPassThruHandler = nil;
 }
 
+- (void)sdl_removeOldButtonHandlersFor:(SDLRPCRequest *)request {
+    for(SDLRPCCorrelationId *key in self.rpcRequestDictionary.allKeys) {
+        if ([self.rpcRequestDictionary[key] isKindOfClass:[SDLAlert class]] && [request isKindOfClass:[SDLAlert class]]) {
+            SDLAlert *oldAlert = (SDLAlert *)self.rpcRequestDictionary[key];
+            SDLAlert *newAlert = (SDLAlert *)request;
+            if(newAlert.softButtons.count != 0) {
+                for (SDLSoftButton *softButton in oldAlert.softButtons) {
+                    if (![[newAlert.softButtons valueForKey:@"softButtonID"] containsObject:softButton.softButtonID]) {
+                        [self.customButtonHandlerMap removeObjectForKey:softButton.softButtonID];
+                    }
+                }
+            }
+        } else if([self.rpcRequestDictionary[key] isKindOfClass:[SDLScrollableMessage class]] && [request isKindOfClass:[SDLScrollableMessage class]]) {
+            SDLScrollableMessage *oldMessage = (SDLScrollableMessage *)self.rpcRequestDictionary[key];
+            SDLScrollableMessage *newMessage = (SDLScrollableMessage *)request;
+            if(newMessage.softButtons.count != 0) {
+                for (SDLSoftButton *softButton in oldMessage.softButtons) {
+                    if (![[newMessage.softButtons valueForKey:@"softButtonID"] containsObject:softButton.softButtonID]) {
+                        [self.customButtonHandlerMap removeObjectForKey:softButton.softButtonID];
+                    }
+                }
+            }
+        } else if([self.rpcRequestDictionary[key] isKindOfClass:[SDLShow class]] && [request isKindOfClass:[SDLShow class]]) {
+            SDLShow *oldShow = (SDLShow *)self.rpcRequestDictionary[key];
+            SDLShow *newShow = (SDLShow *)request;
+            if(newShow.softButtons.count != 0) {
+                for (SDLSoftButton *softButton in oldShow.softButtons) {
+                    // Check if new show's softButton array contains ID of old soft button. (This means button is being updated).
+                    if (![[newShow.softButtons valueForKey:@"softButtonID"] containsObject:softButton.softButtonID]) {
```

Comment:
I think you were correct and I was mistaken that a button could be updated. I tested, and it just replaces all the buttons. This should be unnecessary.